lib/libzrepl/data_conn.c

```diff
@@ -776,10 +782,35 @@ uzfs_zvol_rebuild_dw_replica(void *arg)
 			uzfs_zvol_set_status(zinfo->main_zv,
 			    ZVOL_STATUS_HEALTHY);
 			uzfs_update_ionum_interval(zinfo, 0);
+			/*
+			 * Lets ask io_receiver thread to flush
+			 * all outstanding IOs in taskq
+			 */
+			zinfo->quiesce_done = 0;
+			zinfo->quiesce_requested = 1;
+			quiesce_wait = 1;
 		}
 	}
 	mutex_exit(&zinfo->main_zv->rebuild_mtx);
 
+	/*
+	 * Wait for all outstanding IOs to be flushed
+	 * to disk before making further progress
+	 */
+	while (quiesce_wait) {
```

Comment:
can we make this macro, and, do destroy of clone if rebuild status is healthy in the macro

Reply:
or may be a local variable which will tell to delete the clone, and an assert for HEALTHY status